server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexTemplateService.java

```diff
@@ -922,70 +924,129 @@ private static void validateTemplate(Settings validateSettings, String mappings,
         }
     }
 
-    private void validate(PutRequest request) {
+    private void validate(String name, ComponentTemplate template) {
+        validate(name,
+            template.template().settings(),
+            Collections.emptyList(),
+            Optional.ofNullable(template.template().aliases())
+                .map(aliases -> aliases.values().stream()
+                    .map(aliasMeta -> {
+                        Alias a = new Alias(aliasMeta.alias());
+                        if (aliasMeta.filter() != null) {
+                            a.filter(aliasMeta.filter().string());
+                        }
+                        a.searchRouting(aliasMeta.searchRouting());
+                        a.indexRouting(aliasMeta.indexRouting());
+                        a.isHidden(aliasMeta.isHidden());
+                        a.writeIndex(aliasMeta.writeIndex());
+                        return a;
+                    })
+                    .collect(Collectors.toList()))
+                .orElse(Collections.emptyList()));
+    }
+
+    private void validate(String name, IndexTemplateV2 template) {
+        Optional<Template> maybeTemplate = Optional.ofNullable(template.template());
+        validate(name,
+            maybeTemplate.map(Template::settings).orElse(Settings.EMPTY),
+            template.indexPatterns(),
+            maybeTemplate
+                .map(Template::aliases)
+                .map(aliasMap -> aliasMap.values().stream()
+                    .map(aliasMeta -> {
+                        Alias a = new Alias(aliasMeta.alias());
+                        if (aliasMeta.filter() != null) {
+                            a.filter(aliasMeta.filter().string());
+                        }
+                        a.searchRouting(aliasMeta.searchRouting());
+                        a.indexRouting(aliasMeta.indexRouting());
+                        a.isHidden(aliasMeta.isHidden());
+                        a.writeIndex(aliasMeta.writeIndex());
+                        return a;
+                    })
```

Comment:
Would it make sense to extract this in a separate method and use it in both validate CT/TemplateV2 methods?